modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2146,30 +2160,31 @@ def sort_function(df):
             shuffling_functions,
             sort_function,
         )
-        new_axes = self.axes
+        new_axes = (
+            [None, self._columns_cache]
+            if axis == Axis.ROW_WISE
+            else [self._index_cache, None]
+        )
         new_lengths = [None, None]
         if kwargs.get("ignore_index", False):
-            new_axes[axis.value] = RangeIndex(len(new_axes[axis.value]))
-        else:
-            (
-                new_axes[axis.value],
-                new_lengths[axis.value],
-            ) = self._compute_axis_labels_and_lengths(axis.value, new_partitions)
+            old_axis_value = self.get_axis(axis.value)
+            new_axes[axis.value] = RangeIndex(len(old_axis_value)).set_names(
+                old_axis_value.names
+            )
 
-        new_axes[axis.value] = new_axes[axis.value].set_names(
```

Comment:
This has to happen after the axis labels are computed even when we `ignore_index=True`, right?

Reply:
No. We actually don't need this custom name setting for any of the branches.

- If `ignore_index` is False, then the index names are naturally preserved since we don't drop them anyhow during reshuffling.
- If `ignore_index` is True then the index is being reset to an unnamed range-index

I added two tests for this to check that everything works as expected.